### PR TITLE
expose gps data via webdav

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] https://github.com/nextcloud/documentation/pull/10957
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php [hunk 1]
```diff
@@ -438,6 +439,27 @@ public function handleGetProperties(PropFind $propFind, \Sabre\DAV\INode $node)
 
 					return $sizeMetadata->getValue();
 				});
+
+				$propFind->handle(self::FILE_METADATA_GPS, function () use ($node) {
+					if (!str_starts_with($node->getFileInfo()->getMimetype(), 'image')) {
+						return json_encode((object)[], JSON_THROW_ON_ERROR);
```
Comment (Altahrim) [inline apps/dav/lib/Connector/Sabre/FilesPlugin.php:445]: Maybe just:
```suggestion
						return '{}';
```

**Review by Altahrim — APPROVED** [review]

**Review by susnux — APPROVED** [review]
Do not forget to add the prop to the documentation here:

https://github.com/nextcloud/documentation/blob/master/developer_manual/client_apis/WebDAV/basic.rst#supported-properties

File: apps/dav/lib/Connector/Sabre/FilesPlugin.php (hunk 1 above)
Comment (AndyScherzinger) [reply apps/dav/lib/Connector/Sabre/FilesPlugin.php:445]: I guess it is a one-to-one replication of the code for `files-metadata-size`, so I would keep it as-is.

**Comment by icewind1991** [comment]
Added the pre-loading and de-duplicated some logic

**Review by Altahrim — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Pull request merged**